lib/sen_tweet/bitfeels/stats.ex

```diff
@@ -0,0 +1,65 @@
+defmodule SenTweet.Bitfeels.Stats do
+  @moduledoc """
+  Handles updates and initialization of the sentiment statistics
+  """
+
+  @tweet_types [:extended_tweet, :retweeted_status, :quoted_status, :text]
+  @weight_factors [:tweets, :likes, :retweets]
+
+  def create_stats do
+    for type <- @tweet_types, into: %{}, do: {type, empty_stats()}
+  end
+
+  def update_all_stats(all_stats, measurements, %{tweet_type: type} = metadata) do
+    type = String.to_atom(type)
```

Comment:
Since these type atoms already exist in the system we can use `to_existing_atom`

```suggestion
    type = String.to_existing_atom(type)
```